### PR TITLE
Lnxdsp 928/remove scripts dependency

### PR DESCRIPTION
**Pull request by vasbimpikasadi**
# Changes
This completely removes the need to maintain `lxndsp-scripts` repo any further.

Under the new directory `tools`, the `setup-environment` script from the soon-to-be-defunct repo has been copied over. It is modified to get the sample `local.conf` file gained by yocto when `oe-init-build-env` is ran, and modified it in 3 stages:

1. Add copyright notice, with current year `add_copyright_notice()`
2. Add the horizontal sc5xx edits/additions `sc5xx_universal_edit()`. These were identified after doing a 6-way diff between the platforms that we'll support going forward:
              - sc573-ezkit
              - sc589-mini
              - sc594-som-ez*
              - sc594-som-ez*
    
image
        
3. Add per board changes (ln 252). Group them where possible

# Benefits
This results in the same files that are canned in `lnxdsp-scripts` and can be easily modified. I think having one less repo to maintain and hand-edit yet another conf file, brings the maintenance cost way down.

The biggest benefit however, is including all the changes needed to build in our own meta-layer, instead of more repos. That will come very handy with the continuous efforts of aligning with upstream code/practices.

# Testing
Building with the changes produced no problems in Jenkins: https://www.secad.analog.com/jenkins/dte/job/Linux-ADSP-SC5xx/job/ADSP-SC5xx-builder/ builds #517 through to #543

### CODE REVIEW

File: meta-adi-adsp-sc5xx/recipes-bsp/u-boot/u-boot-adi.inc
Comment (UtsavAgarwalADI) [inline meta-adi-adsp-sc5xx/recipes-bsp/u-boot/u-boot-adi.inc:1]: do we want this to be directed to dev?

**Comment by davidgibsonadi** [comment]
What is the implication here for any customer that has an existing check-out, upgrades and attempts to run their existing scripts/* programs?
Is there anything more than a Release Note "Please remove scripts" that we can do to avoid problems?

**Comment by vasbimpikasadi** [comment]
@davidgibsonadi zero implications. The scripts repo isn't going anywhere and nothing needs to be done for builds that aren't using it, as the .xml will be adjusted to accommodate its removal. One wouldn't even notice a change if they hadn't seen the PR.

File: meta-adi-adsp-sc5xx/recipes-bsp/u-boot/u-boot-adi.inc
Comment (vasbimpikasadi) [reply meta-adi-adsp-sc5xx/recipes-bsp/u-boot/u-boot-adi.inc:1]: yes, dev points to dev and main points to main. This was an oversight fixed!

**Pull request merged**